src/main/java/com/relayrides/pushy/apns/ApnsConnection.java

```diff
@@ -280,13 +281,17 @@ protected void channelRead0(final ChannelHandlerContext context, final RejectedN
 
 			// Regardless of the cause, we ALWAYS want to notify listeners that some sent notifications were not
 			// processed by the gateway (assuming there are some such notifications).
-			final Collection<T> unprocessedNotifications =
-					this.apnsConnection.sentNotificationBuffer.getAllNotificationsAfterSequenceNumber(
-							rejectedNotification.getSequenceNumber());
-
-			if (!unprocessedNotifications.isEmpty()) {
-				if (this.apnsConnection.listener != null) {
-					this.apnsConnection.listener.handleUnprocessedNotifications(this.apnsConnection, unprocessedNotifications);
+			// Unless the sequence number was erroneously reported as 0, in which case we don't know the actual
+			// sequence number, and can't determine what was sent after the bad notification.
+			if (rejectedNotification.getSequenceNumber() != 0) {
```

Comment:
I think this is the best thing we can do here, but let's pause here for a moment of arm-flailing while we consider the implications of this change. That this problem exists at all tells us that, even after the gateway has rejected a notification and is preparing to close, *it will still accept and send notifications received before the connection actually closes.*

I'm preeeeeeetty sure that clobbers what we had previously considered to be a guarantee about the state of a connection and its notifications after a notification is rejected. It seems like this leaves us between a rock and a hard place when it comes to post-rejection notifications; we can either send them again (which risks double-sending) or decide not to re-transmit (which risks never sending at all).

Uuuuuuuuuugh.